code/Test_definitions/population-density-data.feature

```diff
@@ -0,0 +1,277 @@
+        And the response property "$.timedPopulationDensityData[*].startTime" is equal to or later than request body property "$.startDate"
+        And the response property "$.timedPopulationDensityData[*].endTime" is equal to or earlier than request body property "$.endDate"
+        And the response property "$.timedPopulationDensityData[*].cellPopulationDensityData[*].populationDensityData.dataType.datatype" is equal to "LOW_DENSITY" or "DENSITY_ESTIMATION"
+
+    @population_density_data_02_partial_area_success_scenario
+    Scenario: Validate success response for  a partial supported area request
+        Given the request body property "$.area" is set to a valid testing area partially within supported regions
+        And the request body property "$.startDate" is set to a valid testing future date
+        And the request body property "$.endDate" is set to a valid testing future date later than body property "$.startDate"
+        When the request "retrievePopulationDensity" is sent
+        Then the response status code is 200
+        And the response header "Content-Type" is "application/json"
+        And the response header "x-correlator" has same value as the request header "x-correlator"
+        And the response body complies with the OAS schema at "/components/schemas/PopulationDensityResponse"
+        And the response property "$.status" value is "PART_OF_AREA_NOT_SUPPORTED"
+        And the response property "$.timedPopulationDensityData[*].startTime" is equal to or later than request body property "$.startDate"
+        And the response property "$.timedPopulationDensityData[*].endTime" is equal to or earlier than request body property "$.startDate"
+        And there is at least one item in response property "$.timedPopulationDensityData[*].cellPopulationDensityData[*].datatype" equal to "NO_DATA"
+
+    @population_density_data_03_not_supported_area_success_scenario
+    Scenario: Validate success response for unsupported area request
+        Given the request body property "$.area" is set to a valid testing area outside supported regions
+        And the request body property "$.startDate" is set to a valid testing future date
+        And the request body property "$.endDate" is set to a valid testing future date later than body property "$.startDate"
+        When the request "retrievePopulationDensity" is sent
+        Then the response status code is 200
+        And the response header "Content-Type" is "application/json"
+        And the response header "x-correlator" has same value as the request header "x-correlator"
+        And the response body complies with the OAS schema at "/components/schemas/PopulationDensityResponse"
+        And the response property "$.status" value is "AREA_NOT_SUPPORTED"
+        And the response property "$.timedPopulationDensityData" is an empty array
+
+    @population_density_data_04_webhook_success_scenario
```

Comment:
I have an issue on this one because we are not aligned with commonalities for the webjook. It should be sink & sinkCredential. See commonalities or carrier billing API where the pattern has been aligned. I will post an issue.

If you consider that it's tool ate to align not sure keeping the test worthwhile.

Reply:
and here we have the tricky one. This also requires changes in the API yaml, but APi is now in the public release phase. @bigludo7 let me propose something, please confirm:
- we create a new PR to update API yaml
- we update this .feature also with this and other changes
- updates are reported in the #45

Reply:
Hello @jgarciahospital 
This is tricky yes.
 @rartych provided another view: as this API did not use strictly the explicit subscription model, this is is not an issue for him. So perhaps we can change....nothing here for this version (and keep this test) and we aligne for next release.
WDYT? I'm a bit nervous to have all these change...

Reply:
@rartych any proposal about this? Just to ensure that we close it before review. No problem with any of the options, I think we can provide both code proposals now.

Reply:
I would stay with the current definition and discuss changes for version 0.2 of the API. In fact we don't have here Event Notification defined in https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#12-subscription-notification--event , but asynchronous response. There are also no explicit guidelines for that scenario in Commonalities.

Reply:
Ok, let me then update gherkin with only the rest of small comments coming from @bigludo7, and also main yaml with some only editorials, and we are ready for closing the release.

Please @sachinvodafone confirm after that review for been able to merge the test plan.

Reply:
Sorry, making one additional review we find that QoD API suffered similar issue and finally they made the alignement (this [Issue](https://github.com/camaraproject/QualityOnDemand/issues/332) & [PR](https://github.com/camaraproject/QualityOnDemand/pull/335) summarizes). So, just for aligning with https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#instance-based-implicit-subscription, let me propose to roll back to initial plan and:

- Update gherkin to be aligned with async auth
- Update API yaml (I'll create a separate PR moving back to wip version)
- Later we can sync this with the release PR